### PR TITLE
mail/rspamd: allow blacklisting of file extensions; fix bugs

### PR DESCRIPTION
**Pull request by fabianfrz**
Contains some file extensions, which are executable on windows as a default. Please note that sadly js is in the list because it is also used for JScript which has been used to deploy ransomware.

also contains fixes because the generated config files are not readable by the daemon (mode 640 owned by root:wheel and rspamd is nobody:nobody)

### CODE REVIEW

**Comment by fabianfrz** [comment]
CC @mimugmail If you want to test the patch.

File: mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml [hunk 1]
```diff
@@ -333,5 +333,12 @@
         <Required>N</Required>
       </exceptions>
     </surbl>
+    
+    <multimap>
+      <badfileextension type="CSVListField">
+        <Required>Y</Required>
```
Comment (mimugmail) [inline mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml]: Is Required Y correct when someone dont want to use this feature?

Comment (fabianfrz) [reply mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml]: good point will make it a N

File: mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml [hunk 2]
```diff
@@ -333,5 +333,12 @@
         <Required>N</Required>
       </exceptions>
     </surbl>
+
```
Comment (fichtner) [inline mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml]: extra line

File: mail/rspamd/src/opnsense/scripts/rspamd/setup.sh [hunk 3]
```diff
@@ -3,6 +3,12 @@ mkdir -p /var/db/rspamd
 mkdir -p /var/log/rspamd
 mkdir -p /var/run/rspamd
 
+# fix permissions of files generated by configd
+chmod +r /usr/local/etc/rspamd/local.d/*
+chmod o+rx /usr/local/etc/rspamd/local.d
+chown -R nobody /var/log/rspamd
+
```
Comment (fichtner) [inline mail/rspamd/src/opnsense/scripts/rspamd/setup.sh]: extra line

File: mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml (hunk 2 above)
Comment (fabianfrz) [reply mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml]: I do have them everywhere between the sections in the model (every tab is such an block in the Model).
It makes the file easier to read.

Comment (fichtner) [reply mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml]: ok, fine

File: mail/rspamd/src/opnsense/scripts/rspamd/setup.sh (hunk 3 above)
Comment (fabianfrz) [reply mail/rspamd/src/opnsense/scripts/rspamd/setup.sh]: removed the second empty line

File: mail/rspamd/src/opnsense/mvc/app/controllers/OPNsense/Rspamd/forms/settings.xml [hunk 4]
```diff
@@ -14,6 +14,15 @@
                 <help>If you check this box, the local Redis server will be available to the modules (some do not work without it).</help>
             </field>
         </subtab>
+        <subtab id="rspamd-general-multimap" description="Multimap Settings">
+            <field>
+                <id>rspamd.multimap.badfileextension</id>
+                <label>Blocked File Extensions</label>
+                <type>select_multiple</type>
+                <style>tokenize</style>
+                <help>If an attached file has a suffix in this list, the mail will be rejected via an hard reject, which means that the server will be immeadiately informed about the policy violation..</help>
```
Comment (mimugmail) [inline mail/rspamd/src/opnsense/mvc/app/controllers/OPNsense/Rspamd/forms/settings.xml]: Sorry, little late: a hard and two dots at the end.

File: mail/rspamd/src/opnsense/service/templates/OPNsense/Rspamd/bad_file_extensions.map [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{% if helpers.exists('OPNsense.Rspamd.general.enabled') and OPNsense.Rspamd.general.enabled == '1' and helpers.exists('OPNsense.Rspamd.multimap.badfileextension') and OPNsense.Rspamd.multimap.badfileextension != '' %}
+{% for host in OPNsense.Rspamd.multimap.badfileextension.split(',') %}
+{{ host }}
```
Comment (mimugmail) [inline mail/rspamd/src/opnsense/service/templates/OPNsense/Rspamd/bad_file_extensions.map]: Why did you choose "host" .. wouldn't  file or extension make more sense?

Comment (fabianfrz) [reply mail/rspamd/src/opnsense/service/templates/OPNsense/Rspamd/bad_file_extensions.map]: copy and paste

**Comment by fabianfrz** [comment]
@mimugmail added your changes in  511946b - better late than never.

**Pull request merged**